lib/charms/postgresql_k8s/v0/postgresql.py

```diff
@@ -420,19 +421,39 @@ def update_user_password(self, username: str, password: str) -> None:
             if connection is not None:
                 connection.close()
 
+    def is_restart_pending(self) -> bool:
+        """Query pg_settings for pending restart."""
+        connection = None
+        try:
+            with self._connect_to_database() as connection, connection.cursor() as cursor:
+                cursor.execute("SELECT COUNT(*) FROM pg_settings WHERE pending_restart=True;")
+                return cursor.fetchone()[0] > 0
+        except psycopg2.OperationalError:
+            logger.warning("Failed to connect to PostgreSQL.")
+            return False
+        except psycopg2.Error as e:
+            logger.error(f"Failed to check if restart is pending: {e}")
+            return False
+        finally:
+            if connection:
+                connection.close()
+
     @staticmethod
     def build_postgresql_parameters(
-        profile: str, available_memory: int
+        profile: str, available_memory: int, limit_memory: Optional[int] = None
     ) -> Optional[dict[str, str]]:
         """Builds the PostgreSQL parameters.
 
         Args:
             profile: the profile to use.
             available_memory: available memory to use in calculation in bytes.
+            limit_memory: (optional) limit memory to use in calculation in bytes.
 
         Returns:
             Dictionary with the PostgreSQL parameters.
         """
+        if limit_memory:
```

Comment:
This should be fixed in the future since it's not limiting the amount of memory used by postgres, but using the limit value in place of the available one (case smaller).